### PR TITLE
fix start_vm.sh

### PR DESCRIPTION
**Pull request by bennyz**
If a VM was previously defined and removed, its DHCP entry will remain and will prevent a new VM from getting the desired IP address

### CODE REVIEW

File: contrib/scripts/start_vm.sh [hunk 1]
```diff
@@ -26,15 +26,29 @@ if [  "${fc_host_status}" == 'running' ]; then
     exit 0
 fi
 
-MAC_ADDRESS=$(virsh dumpxml "${VM_NAME}" | grep "mac address" | awk -F\' '{ print $2}')
-echo "Setting IP address to ${VM_IP} for MAC address ${MAC_ADDRESS}"
+mac_address=$(virsh dumpxml "${VM_NAME}" | grep "mac address" | awk -F\' '{ print $2}')
+echo "Setting IP address to ${VM_IP} for MAC address ${mac_address}"
```
Comment (douek) [inline contrib/scripts/start_vm.sh:30]: It is not clear from the message of you are setting the IP or trying to set the ip/

File: contrib/scripts/start_vm.sh [hunk 2]
```diff
@@ -26,15 +26,29 @@ if [  "${fc_host_status}" == 'running' ]; then
...
 
-# TODO: check both IP and MAC, since this wouldn't work if we use a new VM after previously defining one
-if virsh net-dumpxml "${LIBVIRT_NETWORK}" | grep -q "${VM_NAME}"; then
+if [[ "$xml_entry" == "$quote_stripped_entry" ]]; then
     echo "IP address is already configured"
 else
+    if [[ -n "$existing_entry" ]]; then
+        existing_mac=$(virsh net-dumpxml --network default | grep fchost | awk -F\' '{ print $2}')
+        if [[ "$existing_mac" != "$mac_address" ]]; then
+            echo "Removing existing entry..."
```
Comment (douek) [inline contrib/scripts/start_vm.sh:42]: Are you sure you want to remove the existed entry? It cannot be undo. Maybe we can add an argument for force or an input from the user to make sure.
If no IP has been given as an argument, maybe the default should be a free IP and not a specific one (192.168.122.45)

File: contrib/scripts/start_vm.sh [hunk 3]
```diff
@@ -26,15 +26,29 @@ if [  "${fc_host_status}" == 'running' ]; then
     exit 0
 fi
 
-MAC_ADDRESS=$(virsh dumpxml "${VM_NAME}" | grep "mac address" | awk -F\' '{ print $2}')
-echo "Setting IP address to ${VM_IP} for MAC address ${MAC_ADDRESS}"
+mac_address=$(virsh dumpxml "${VM_NAME}" | grep "mac address" | awk -F\' '{ print $2}')
```
Comment (douek) [inline contrib/scripts/start_vm.sh:29]: If the VM is already installed you don't need to do through the process to lease an ip for the mac address (it should already have one) and we just want to start the vm.

**Comment by bennyz** [comment]
Switched to terraform, no longer relevant

**Pull request closed without merging**